lib/reservations.rb

```diff
@@ -0,0 +1,40 @@
+require 'date'
+
+module Hotel
+  class Reservation
+    ROOMCOST = 200
+    attr_accessor :check_in, :check_out, :room_num
+    #things that only have to deal with SINGLE ROOM, one reservation at a time
+    #monitors overlapping
+    def initialize(check_in, check_out, room_num)
+      @check_in = check_in
+      @check_in = check_out
+      @room_num = room_num
+        if @room_num > 20 || @room_number < 1
+          raise ArgumentError.new("Number must be between 1 - 20")
+        end
+    end
+
+    def reserve_room(check_in, check_out, room_num)
+      #reserves a room for a given date range
+      in = Date.parse(check_in)
```

Comment:
You are using a keyword here, `in` means something special in Ruby so you can't use it for a variable name.